x/auth/ante/circuit.go

```diff
@@ -0,0 +1,22 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type circuitBreakerDecorator struct {
+	keeper CircuitBreakerKeeper
+}
+
+func (d *circuitBreakerDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
```

Comment:
This should not be a reference...it might be why you're seeing some `nil` pointer deferences in tests.
```suggestion
func (d circuitBreakerDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
```